sdk/storage/azure-storage-blob/tests/test_blob_access_conditions.py

```diff
@@ -715,6 +715,20 @@ def test_get_blob_properties_with_if_modified(self, resource_group, location, st
         self.assertEqual(properties.size, 11)
         self.assertEqual(properties.lease.status, 'unlocked')
 
+    @GlobalStorageAccountPreparer()
+    def test_if_blob_exists(self, resource_group, location, storage_account, storage_account_key):
+        bsc = BlobServiceClient(self.account_url(storage_account, "blob"), storage_account_key, connection_data_block_size=4 * 1024)
+        self._setup()
+        self._create_container_and_block_blob(
+            self.container_name, 'blob1', b'hello world', bsc)
+        test_datetime = (datetime.utcnow() -
+                         timedelta(minutes=15))
+        # Act
+        blob = bsc.get_blob_client(self.container_name, 'blob1')
+        blob_version_id = blob.get_blob_properties().get("version_id")
```

Comment:
let's 

1. create blob first, you will get a version id, make sure the version_id is not empty.
2. overwrite the existing blob, so currently this blob is the root blob.
3. check if the overwritten blob exists using the version_id
4. check if the root blob exists

also after you enable checking snapshot, you also need to test the snapshot.